network/wsNetwork.go

```diff
@@ -736,24 +736,57 @@ func (wn *WebsocketNetwork) getForwardedConnectionAddress(header http.Header) (i
 	return
 }
 
-func (wn *WebsocketNetwork) checkHeaders(header http.Header, addr string, forwardedAddr net.IP) (ok bool, otherTelemetryGUID string, otherPublicAddr string, otherInstanceName string) {
-	ok = false
+type checkHeadersError interface {
+	error
+	Type() string
+}
+
+type mismatchingVersionError struct {
+	version string
+}
+
+func (e *mismatchingVersionError) Error() string {
+	return fmt.Sprintf("Requested version %s = %s mismatches server version", ProtocolVersionHeader, e.version)
+}
+
+func (e *mismatchingVersionError) Type() string {
+	return "mismatching protocol version"
+}
+
+type matchingRandomIDError struct {
+	randomID string
+}
+
+func (e *matchingRandomIDError) Error() string {
+	if e.randomID == "" {
+		return fmt.Sprintf("Missing random ID header %s", NodeRandomHeader)
+	}
+	return fmt.Sprintf("Requested node random ID %s = %s matches server's random ID", NodeRandomHeader, e.randomID)
+}
+
+func (e *matchingRandomIDError) Type() string {
+	return "matching node random ID"
+}
+
+func (wn *WebsocketNetwork) checkHeaders(header http.Header, addr string, forwardedAddr net.IP) (otherTelemetryGUID string, otherPublicAddr string, otherInstanceName string, err checkHeadersError) {
 	otherTelemetryGUID = ""
 	otherPublicAddr = ""
 	otherVersion := header.Get(ProtocolVersionHeader)
 	if otherVersion != ProtocolVersion {
-		wn.log.Warnf("new peer %#v version mismatch, mine=%#v theirs=%#v, headers %#v", addr, ProtocolVersion, otherVersion, header)
-		return
-	}
-	otherGenesisID := header.Get(GenesisHeader)
-	if len(otherGenesisID) > 0 && wn.GenesisID != otherGenesisID {
-		wn.log.Warnf("new peer %#v genesis mismatch, mine=%#v theirs=%#v, headers %#v", addr, wn.GenesisID, otherGenesisID, header)
+		wn.log.Warnf("new peer %s version mismatch, mine=%s theirs=%s, headers %#v", addr, ProtocolVersion, otherVersion, header)
```

Comment:
I think this should be an Info level log entry.  This is going to happen when someone doesn't upgrade their node in a timely fashion.

Reply:
make sense; I'll update the code.